### PR TITLE
Add support for a maxLines property on Paragraph.

### PR DESCRIPTION
**Pull request by mpcomplete**
Also fix 'ellipsis' overflow handling to work for multiline text.

Fixes https://github.com/flutter/flutter/issues/7271

### CODE REVIEW

File: lib/ui/text.dart [hunk 1]
```diff
@@ -491,8 +501,9 @@ class ParagraphStyle {
              'lineCount: ${   _encoded[0] & 0x10 == 0x10 ? _encoded[4]                      : "unspecified"}, '
              'fontFamily: ${  _encoded[0] & 0x20 == 0x20 ? _fontFamily                      : "unspecified"}, '
              'fontSize: ${    _encoded[0] & 0x40 == 0x40 ? _fontSize                        : "unspecified"}, '
-             'lineHeight: ${  _encoded[0] & 0x80 == 0x80 ? "${_lineHeight}x"                : "unspecified"}, '
-             'ellipsis: ${    _encoded[0] & 0x100 == 0x100 ? "\"$_ellipsis\""                : "unspecified"}'
+             'lineHeight: ${  _encoded[0] & 0x80 == 0x80 ? _lineHeight                      : "unspecified"}, '
```
Comment (abarth) [inline lib/ui/text.dart]: Looks like you lost the `x` on line height.

File: lib/ui/text.dart [hunk 2]
```diff
@@ -634,7 +645,7 @@ class ParagraphConstraints {
     return typedOther.width == width;
   }
 
-  int get hashCode => width.hashCode;
+  int get hashCode => hashValues(width);
```
Comment (abarth) [inline lib/ui/text.dart]: There's no reason to go through hashValues for just one value.  The hashCode of the value itself is fine.

**Comment by abarth** [comment]
Looks plausible to me but @jason-simmons should take a look.

**Comment by jason-simmons** [comment]
LGTM

**Pull request merged**